### PR TITLE
Write conclusion

### PR DESCRIPTION
**Pull request by daffidwilde**
Write a conclusion which summarises the research reported in the thesis, as well as its contributions to literature and potential avenues of future work.

### CODE REVIEW

File: chapters/conclusion.tex [hunk 1]
```diff
@@ -1 +1,217 @@
...
+review, Chapter~\ref{chp:kmodes} offered a novel initialisation to the
+\(k\)-modes algorithm which made use of game theory. The new initialisation
+extended a commonly used method, but replaced its greedy component with a
+solvable matching game. In the evaluative section of this chapter, traditional
+assessment techniques suggested that the new method improved upon the original,
+and so the original was discarded.
+
+However, the new method did not consistently outperform another well-known
+initialisation. To better understand the conditions under which either of the
+remaining initialisations would succeed, a similar competitive setting to
```
Comment (drvinceknight) [inline chapters/conclusion.tex:48]:
```suggestion
remaining initialisations would succeed, a similar setting to
```

File: chapters/conclusion.tex [hunk 2]
```diff
@@ -1 +1,217 @@
...
+literature in three ways. First, the estimation of queuing parameters via the
+Wasserstein distance has expanded a relatively scarce aspect of queuing
+research. Second, by making COPD the subject of the methodology, the framework
+has added to a body of literature surrounding a condition that is vital to
+understand given its prevalence, as well as its links to deprivation and
+comorbidity. Lastly, the framework provided a solution to the common issue of
+data availability in modern operational research. By combining the various
+individual methods, valuable insights were extracted from a relatively
+unsophisticated data source, which is a result seldom seen in operational
+research.
```
Comment (drvinceknight) [inline chapters/conclusion.tex:117]: Nice. :+1:

File: chapters/conclusion.tex [hunk 3]
```diff
@@ -1 +1,217 @@
...
+
+In addition to the work directly included in the chapters of this thesis, the
+research associated with this thesis has resulted in the production of numerous
+auxiliary research items. These include several well-developed pieces of
+research software, and a number of useful, publicly available datasets for
+clustering and healthcare modelling.
```
Comment (drvinceknight) [inline chapters/conclusion.tex:123]: Can we point at them (listed in brackets is fine, or perhaps this needs to be a standalone page with everything listed, I forget if you have that already?).

File: chapters/conclusion.tex [hunk 4]
```diff
@@ -1 +1,217 @@
...
+
+\section{Further work}
+
+\subsection*{EDO as a data synthesiser}
+
+As demonstrated in the case study in Chapter~\ref{chp:edo} and the closing
+section of Chapter~\ref{chp:kmodes}, the EDO method is capable of facilitating
+richer insights into an algorithm's performance. Having said that, a limitation
+of the method is that there is no standardised way to guarantee relationships
+between different columns in a dataset, or the families passed to EDO.
```
Comment (drvinceknight) [inline chapters/conclusion.tex:134]:
```suggestion
between different columns in a dataset (\(\mathcal P\), or the families passed to EDO.
```

File: chapters/conclusion.tex [hunk 5]
```diff
@@ -1 +1,217 @@
...
+fitted to the existing dataset. In this case, the technical aspects of an
+individual's representation would need adjusting to accommodate this change.
+Likewise, the crossover and mutation processes would require some changes to
+account for the lack of distinct distribution families.
+
+A Python implementation of copulas for data synthesis exists~\cite{copulas}, and
+incorporating this as a dependency of the \edo\ library would reduce the work
+required to implement this feature. Studying the impact of copulas in EDO would
+provide a valuable opportunity to demonstrate the capabilities of EDO as a fully
+fledged data synthesis method.
```
Comment (drvinceknight) [inline chapters/conclusion.tex:165]: That's neat!

Comment (drvinceknight) [reply chapters/conclusion.tex:165]: You should open this up as an issue on https://github.com/daffidwilde/edo/issues

**Review by drvinceknight — CHANGES_REQUESTED** [review]
This all looks good @daffidwilde, the only thing you could perhaps add in is some of the reflection on the general direction of the research (essentially what you had put in in your introduction?).

Not sure if it's necessary. Whatever you feel happier with.

File: chapters/conclusion.tex (hunk 3 above)
Comment (daffidwilde) [reply chapters/conclusion.tex:123]: I don't have one for the datasets, but I do list the DOI names for them in the introduction. The software packages are summarised in the dissemination.

Do you think another page would be helpful?

Comment (drvinceknight) [reply chapters/conclusion.tex:123]: It's probably fine to have those two spots in which case can you just point to them here? Something like:

```
research software (listed in \ref{dissemination}, and a number of useful, publicly available datasets (listed in Section \ref{}).
```

Comment (daffidwilde) [reply chapters/conclusion.tex]: Cool cool. Added in those references in ac10058

**Comment by daffidwilde** [comment]
> you could perhaps add in is some of the reflection on the general direction of the research

Are there any particular examples from the introduction you liked?

File: chapters/conclusion.tex (hunk 5 above)
Comment (daffidwilde) [reply chapters/conclusion.tex:165]: Will do! 😄

Comment (daffidwilde) [reply chapters/conclusion.tex:165]: Done.

**Comment by drvinceknight** [comment]
> Are there any particular examples from the introduction you liked @drvinceknight?

Nothing specifically @daffidwilde, I thought the entire reflective part was pretty good (just not in the right place). If **you** wanted to include it, you could put in in this chapter under `\section{Reflections on research direction}` (or something like that).

**Comment by drvinceknight** [comment]
> If **you** wanted to include it

If you don't want to include it, that's also 100% fine by me :)

**Comment by daffidwilde** [comment]
I think it's a good idea to have it in there. I've added a section in 01eba00 that basically takes those paragraphs and scrubs them up a bit.

File: chapters/conclusion.tex [hunk 6]
```diff
@@ -1 +1,283 @@
...
+
+First, the vast majority of publications that introduced a novel machine
+learning method contained a boilerplate evaluation section. In each article, the
+proposed method would be pitted against a few of its contemporaries by comparing
+a handful of metrics on a handful of datasets. Typically, these metrics and
+datasets would be taken from a small pool which was relevant to that technique;
+this makes sense. There are appropriate measures for various techniques, and it
+is important to make comparisons relative to a fixed point.
+
+However, this approach to algorithm evaluation exhibited two issues: one, a lack
```
Comment (drvinceknight) [inline chapters/conclusion.tex:154]:
```suggestion
This approach to algorithm evaluation exhibited two issues: one, a lack
```

File: chapters/conclusion.tex [hunk 7]
```diff
@@ -1 +1,283 @@
...
+incongruence between the power of the evaluation process and the conclusions
+drawn from that process. Often, a method would be deemed `state-of-the-art' or
+`better' based exclusively on a process that offered little insight into the
+actual quality of that method. This reliance on a narrow assessment process
+prompted research into how else an algorithm could be evaluated objectively,
+directly leading to the work in Chapter~\ref{chp:edo}.
+
+The second observation was that the ethics of data and machine learning
+algorithms were being discussed, but the discourse appeared separately from the
+machine learning publications. An exception to this was the development fair
```
Comment (drvinceknight) [inline chapters/conclusion.tex:165]:
```suggestion
machine learning publications. An exception to this was the development of fair
```

File: chapters/conclusion.tex [hunk 8]
```diff
@@ -1 +1,283 @@
...
+would have to prioritise extracting as much useful information from the data at
+hand. Simultaneously, \ctmuhb\ provided further administrative datasets which
+related to the members of their population who suffer from the respiratory
+condition, COPD. An internal report by NHS Wales found that \ctmuhb\ had the
+highest prevalence of the condition out of all the Welsh health boards. Given
+the known links between COPD, socioeconomic deprivation and the coincidence of
+multiple comorbidities, the objective of this project was revised to focus on
+understanding the needs of that population.
+
+Each of these concerns culminated in the methodology presented in
```
Comment (drvinceknight) [inline chapters/conclusion.tex:184]: I wonder if we can use a softer word than "concerns", perhaps "challenges"?

```suggestion
Overcoming each of these challenges culminated in the methodology presented in
```

**Review by drvinceknight — COMMENTED** [review]
I think it reads really nice @daffidwilde :+1:

File: chapters/conclusion.tex (hunk 8 above)
Comment (daffidwilde) [reply chapters/conclusion.tex:184]: A good point well made. I'll make these changes now 👍

**Pull request merged**